Monkey patching the #helper method causing some hooks not to load.

jmandala opened the pull request:
Monkey patching the #helper method causing some hooks not to load. This error is seen in Spreecommerce.
Unfortunately, I don't enough about rails internals to fix the problem, aside from defaulting to the normal
behavior, and restarting spork when helper files change.

Is there anything I can do to help determine an appropriate solution?

timcharper commented:
If you feel motivated, you could write a cucumber feature that shows the problem fail. There's one there already that creates a bare rails app, populates a file, and then tests it, you could use it as a template. :)

jmandala commented:
That sounds like a useful exercise. I'll give it a try.

jmandala commented:
I'm pretty sure this problem was a user problem, and a not the fault of the source code.

I'm using Spork+Cucumber+Rails, and my Spork.prefork block didn't require 'cucumber/rails'. That caused the environment not to be setup fully for my subsequent configuration and tests. I guess that's what I get for looking for cookie cutter solutions! Once I dug into the source, it started to make sense. 

FWIW, here's my Spork.prefork

Spork.prefork do
  ENV["RAILS_ENV"] ||= "cucumber"
  require File.expand_path(File.dirname(**FILE**) + '/../../config/environment')

  require 'cucumber'
  require 'cucumber/rails'
  require 'cucumber/formatter/unicode' # Comment out this line if you don't want Cucumber Unicode support
  require 'rspec/rails'
  require 'cucumber/rails/rspec'
end

Thanks so much for taking the time to reply and motivate me to look a little further!

jmandala committed:
Monkey patching the #helper method causing some hooks not to load. This error is seen in Spreecommerce.
Unfortunately, I don't enough about rails internals to fix the problem, aside from defaulting to the normal
behavior, and restarting spork when helper files change.

timcharper commented:
Well nice work finding the problem :) you made me proud!
